docs/transforms.md

```diff
@@ -25,20 +25,68 @@ A transform consists of 4 parts: type, name, matcher, and transformer. Transform
 
 ## Defining Custom Transforms
-You can define custom transforms with the [`registerTransform`](api.md#registertransform). Style Dictionary adds some [default metadata](properties.md?id=default-property-metadata) to each property/token to provide context that may be useful for some transforms.
+You can define custom transforms with the [`registerTransform`](api.md#registertransform). Style Dictionary adds some [default metadata](tokens.md?id=default-design-token-metadata) to each design token to provide context that may be useful for some transforms.
+
+## Transitive Transforms
+Starting in version 3.0, you can define transitive transforms which allow you to transform a referenced value. Normally, value transforms only transform non-referenced values and because transforms happen before references are resolved, the transformed value is then used to resolve references.
+
+```javascript
+const StyleDictionary = require('style-dictionary');
+
+StyleDictionary.registerTransform({
+  type: `value`,
+  transitive: true,
+  name: `myTransitiveTransform`,
+  matcher: (token) => {},
+  transformer: (token) => {
+    // token.value will be resolved and transformed at this point
+  }
+})
+```
+
+There is one thing to be mindful of with transitive transforms. The token's value will be resolved and *transformed* already at the time the transitive transform. What happens is Style Dictionary will transform and resolve values iteratively. First it will transform any non-referenced values, then it will resolve any references to non-referenced values, then it will try to transform any non-referenced values, and so on. Let's take a look at an example:
+
+```json
+{
+  "color": {
+    "red": { "value": "#f00" },
+    "danger": { "value": "{color.red}" },
+    "error": { "value": "{color.danger}" }
+  }
+}
+```
+
+Style dictionary will first transform the value of `color.red`, then resolve `color.danger` to the transformed `color.red` value. Then it will transform `color.danger` and resolve `color.error` to the transformed `color.danger`. Finally, it will transform `color.error` and see that there is nothing left to transform or resolve.
+
+This allows you to modify a reference that modifies another reference. For example:
+
+```json
+{
+  "color": {
+    "red": { "value": "#f00" },
+    "danger": { "value": "{color.red}", "darken": 0.75 },
+    "error": { "value": "{color.danger}", "darken": 0.5 }
+  }
+}
+```
+
+Using a custom transitive transform you could have `color.danger` darken `color.red` and `color.error` darken `color.danger`. The pre-defined transforms are *not transitive* to be backwards compatible and not break anything.
```

Comment:
> The pre-defined transforms are *not transitive* to be backwards compatible and not break anything.

The pre-defined transforms are *not transitive* to be backwards compatible with Style Dictionary 2.0 - an upgrade should not cause breaking changes.